terraform/modules/spack/s3.tf

```diff
@@ -0,0 +1,41 @@
+resource "aws_s3_bucket_policy" "protected_binaries_restricted" {
+  for_each = var.s3_bucket_policies
+
+  bucket = "${each.key}"
+
+  policy = jsonencode({
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Sid": "PublicRead",
+        "Effect": "Allow",
+        "Principal": "*",
+        "Action": "s3:GetObject",
+        "Resource": "arn:aws:s3:::${each.key}/*"
+      },
+      {
+        "Sid": "DenyReadToProtected",
+        "Effect": "Deny",
+        "Principal": {
+          "AWS": "*"
+        },
+        "Action": "s3:GetObject",
+        "Resource": each.value.resources,
+        "Condition": {
+          "ArnNotLike": {
+            "aws:PrincipalArn": "${each.value.allowed}"
+          }
+        }
+      },
+      {
+        "Sid": "AllowReadToProtected",
+        "Effect": "Deny",
+        "Principal": {
+          "AWS": "${each.value.allowed}"
+        },
+        "Action": "s3:GetObject",
+        "Resource": each.value.resources
+      }
+    ]
+  })
+}
```

Comment:
If this is only applicable to production, could we have these be a part of the root `production` module instead of the `spack` module? i.e. similar to how we have [these IAM roles and policies](https://github.com/spack/spack-infrastructure/blob/main/terraform/production/service_account_roles.tf) defined outside of the `spack` module since they are only used in production.